app/src/main/java/edu/wpi/first/shuffleboard/app/MainWindowController.java

```diff
@@ -277,29 +276,32 @@ public void load() {
   @SuppressWarnings("unchecked")
   @FXML
   public void showPrefs() {
-    // Create the property sheet
-    PropertySheet propertySheet = new WidgetPropertySheet(AppPreferences.getInstance().getProperties());
+    PropertySheet propertySheet
+        = new WidgetPropertySheet(AppPreferences.getInstance().getProperties());
+
     propertySheet.setModeSwitcherVisible(false);
     propertySheet.setSearchBoxVisible(false);
     propertySheet.setMode(PropertySheet.Mode.NAME);
-    StackPane pane = new StackPane(propertySheet);
-    pane.setPadding(new Insets(8));
-    Scene scene = new Scene(pane);
-    EasyBind.listBind(scene.getRoot().getStylesheets(), root.getStylesheets());
-
-    Stage stage = new Stage();
-    stage.setScene(scene);
-    stage.initModality(Modality.APPLICATION_MODAL);
-    stage.addEventHandler(KeyEvent.KEY_PRESSED, event -> {
-      if (event.getCode() == KeyCode.ESCAPE) {
-        stage.close();
-      }
-    });
-    stage.setTitle("Shuffleboard Preferences");
-    stage.sizeToScene();
-    stage.setResizable(false);
-    stage.requestFocus();
-    stage.showAndWait();
+
+    Dialog<Boolean> dialog = new Dialog<>();
+    dialog.getDialogPane().setContent(propertySheet);
+    dialog.initModality(Modality.APPLICATION_MODAL);
+    dialog.getDialogPane().getButtonTypes().addAll(ButtonType.CANCEL, ButtonType.OK);
+    dialog.setTitle("Shuffleboard Preferences");
+    dialog.setResizable(true);
+    dialog.setResultConverter(button -> !button.getButtonData().isCancelButton());
+    if (dialog.showAndWait().orElse(false)) {
+      propertySheet.getItems().stream()
+          .map(item -> (WidgetPropertySheet.PropertyItem) item)
+          .map(WidgetPropertySheet.PropertyItem::getObservableValue)
+          .filter(Optional::isPresent)
+          .map(Optional::get)
```

Comment:
god I hate that java's Optional api doesn't implement the stream api... these three calls could just be one flatMap

Reply:
It's pretty easy to implement a version of Optional that implements `Stream`. Or we could use Kotlin and add an extension method 😉